Dockerfile

```diff
@@ -1,34 +1,53 @@
 FROM rocker/tidyverse:latest
 ENV NB_USER rstudio
 ENV NB_UID 1000
+ENV VENV_DIR /srv/venv
+ENV PATH ${VENV_DIR}/bin:$PATH
+
+# The `rsession` binary that is called by nbrsessionproxy to start R doesn't seem to start
+# without this being explicitly set
+ENV LD_LIBRARY_PATH /usr/local/lib/R/lib
+
 ENV HOME /home/${NB_USER}
+WORKDIR ${HOME}
 
 RUN apt-get update && \
-    apt-get -y install python3-pip && \
+    apt-get -y install python3-venv python3-dev && \
+    apt-get purge && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Create a venv dir owned by unprivileged user & set up notebook in it
+# This allows non-root to install python libraries if required
+RUN mkdir -p ${VENV_DIR} && chown -R ${NB_USER} ${VENV_DIR}
+
+USER ${NB_USER}
+RUN python3 -m venv ${VENV_DIR} && \
     pip3 install --no-cache-dir \
          notebook==5.2 \
          git+https://github.com/jupyterhub/nbrsessionproxy.git@6eefeac11cbe82432d026f41a3341525a22d6a0b \
          git+https://github.com/jupyterhub/nbserverproxy.git@5508a182b2144d29824652d8977b32302517c8bc && \
-    apt-get purge && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/* && \
-    R --quiet -e "devtools::install_github('IRkernel/IRkernel')"
+    jupyter serverextension enable --sys-prefix --py nbserverproxy && \
+    jupyter serverextension enable --sys-prefix --py nbrsessionproxy && \
+    jupyter nbextension install    --sys-prefix --py nbrsessionproxy && \
+    jupyter nbextension enable     --sys-prefix --py nbrsessionproxy
 
 
-## These commands must be run in $HOME by $NB_USER, with $NB_UID already set
-## note: for some reason, just doing in a root run command with su ${NB_USER}
-USER ${NB_USER}
-WORKDIR ${HOME} 
+RUN R --quiet -e "devtools::install_github('IRkernel/IRkernel')"
+
+# For some reason, R doesn't seem to want to interpret PATH from the environment!
+#
+#   rstudio@32b818e938c3:~$ echo $PATH
+#   /srv/venv/bin:/usr/lib/rstudio-server/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+#   rstudio@32b818e938c3:~$ R --quiet -e 'Sys.getenv("PATH")'
+#   > Sys.getenv("PATH")
+#   [1] "/usr/lib/rstudio-server/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+#
+# This causes it to never find the `jupyter` script in ${VENV_DIR}/bin.
+# This is the only hack that seems to work.
+RUN R --quiet -e "Sys.setenv(PATH='${PATH}'); IRkernel::installspec(prefix='${VENV_DIR}')"
```

Comment:
yeah, so this is because PATH is getting set in .Renviron https://github.com/rocker-org/rocker-versioned/blob/master/rstudio/Dockerfile#L52.  

I'm open to advice here -- this is technically I think an RStudio issue.  Normally (e.g. in `r-ver` base image), PATH does not get set in .Renviron and things behave the way you expect them to on the R terminal, getting PATH from the system environment.  However, we added this explicit setting because RStudio decided not to use the system path at all, but was just setting it's own path, so almost nothing was available.  So the line above is something of a hack to make more system binaries callable from RStudio.  

I should probably dig into RStudio docs and figure out a better solution than hardwiring PATH in .Renviron....

Reply:
I just copied your hack in here again, which probably works for now?